app/recordtransfer/models.py

```diff
@@ -202,6 +195,23 @@ def expires_within(self, minutes: int) -> bool:
             return False
         return expiry_time < timezone.now() + timezone.timedelta(minutes=minutes)
 
+    def expire(self) -> None:
+        """Set the status of this session to EXPIRED, but only if the current status is
+        CREATED or UPLOADING.
+
+        Raises:
+            ValueError: If the current status is not CREATED or UPLOADING.
+        """
+        if self.status not in (self.SessionStatus.CREATED, self.SessionStatus.UPLOADING):
+            raise ValueError(
+                f"Cannot mark session {self.token} as expired because the session status "
+                f"is {self.status} and not {self.SessionStatus.CREATED} or "
+                f"{self.SessionStatus.UPLOADING}"
+            )
+
+        self.status = self.SessionStatus.EXPIRED
+        self.save()
```

Comment:
We may as well move calling this function inside `expire` so that we never forget to remove temp uploads when expiring an upload session:

```suggestion
        self.remove_temp_uploads()
        self.status = self.SessionStatus.EXPIRED
        self.save()
```